File: samples/Ruby/number-reversal-game.rb
ary = (1..9).to_a
ary.shuffle! while ary == ary.sort
score = 0
until ary == ary.sort
  print "#{ary.inspect} -- How many digits to reverse? "
  num = gets.to_i  # should validate input
  ary[0, num] = ary[0, num].reverse
  score += 1
end
p ary
puts "Your score: #{score}"

File: samples/Mathematica/number-reversal-game.m
Module[{array = Range@9, score = 0},
 While[array == Range@9, array = RandomSample@Range@9];
 While[array != Range@9,
  Print@array; (array[[;; #]] = Reverse@array[[;; #]]) &@
   Input["How many digits would you like to reverse?"]; score++];
 Print@array; Print["Your score:", score]]
